src/main/java/uk/gov/companieshouse/confirmationstatementapi/eligibility/impl/CompanyTypeValidation.java

```diff
@@ -0,0 +1,26 @@
+package uk.gov.companieshouse.confirmationstatementapi.eligibility.impl;
+
+import java.util.Set;
+
+import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
+import uk.gov.companieshouse.confirmationstatementapi.eligibility.EligibilityFailureReason;
+import uk.gov.companieshouse.confirmationstatementapi.eligibility.EligibilityRule;
+import uk.gov.companieshouse.confirmationstatementapi.exception.EligibilityException;
+
+public class CompanyTypeValidation implements EligibilityRule<CompanyProfileApi> {
```

Comment:
Similarly to other comments, may be best to rename this CompanyTypeValidationNoCS01
Just so glancing at the names of the rule classes you get an idea specifically what they are for